### PR TITLE
Implements absolute value for integer circuit literals

### PR DESCRIPTION
**Pull request by howardwu**
## Motivation

Implements absolute value for integer circuit literals

## Test Plan

Tests the absolute value implementation for all integer types, on all modes.

### CODE REVIEW

File: circuits/environment/src/traits/integers.rs [hunk 1]
```diff
@@ -166,18 +168,78 @@ pub(super) mod integer_type {
     impl IntegerType for u64 {}
     impl IntegerType for u128 {}
 
-    macro_rules! checked_impl {
```
Comment (d0cd) [inline circuits/environment/src/traits/integers.rs:170]: Clean up macros for checked and wrapping impls

File: circuits/environment/src/traits/integers.rs [hunk 2]
```diff
@@ -293,18 +355,48 @@ pub(super) mod integer_type {
 
         /// Returns the name of the integer type as a string slice. (i.e. "u8")
         fn type_name() -> &'static str;
+
+        /// Returns the absolute value of the integer type.
```
Comment (d0cd) [inline circuits/environment/src/traits/integers.rs]: Remove `absolute` from `IntegerProperties`

File: circuits/types/integers/src/abs_wrapped.rs [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+        // Check the 0 case.
+        let name = format!("Abs: {} zero", mode);
+        check_abs(&name, I::zero(), mode, num_constants, num_public, num_private, num_constraints);
+
+        // Check the 1 case.
+        let name = format!("Abs: {} one", mode);
+        check_abs(&name, I::one(), mode, num_constants, num_public, num_private, num_constraints);
+
+        // Check the I::MIN (checked) case.
```
Comment (d0cd) [inline circuits/types/integers/src/abs_wrapped.rs]: Rename "(checked)" to "(wrapped)"

File: circuits/types/integers/src/neg.rs [hunk 4]
```diff
@@ -47,14 +47,8 @@ mod tests {
...
 
-    fn check_unsigned_neg_halts<I: IntegerType + std::panic::UnwindSafe>(mode: Mode, value: I) {
-        let candidate = Integer::<Circuit, I>::new(mode, value);
-        let operation = std::panic::catch_unwind(|| candidate.neg());
-        assert!(operation.is_err());
-    }
-
     #[rustfmt::skip]
-    fn check_neg<I: IntegerType + std::panic::UnwindSafe + Neg<Output = I> >(
+    fn check_neg<I: IntegerType + std::panic::RefUnwindSafe + std::panic::UnwindSafe + Neg<Output = I> >(
```
Comment (d0cd) [inline circuits/types/integers/src/neg.rs]: May not need RefUnwindSafe, same in AbsWrapped, AbsChecked.

**Review by d0cd — CHANGES_REQUESTED** [review]

**Review by d0cd — COMMENTED** [review]
Ready for re-review.

**Comment by howardwu** [comment]
LGTM

**Pull request merged**